src/tpm2_wrap.c

```diff
@@ -2873,7 +2919,21 @@ int wolfTPM2_NVReadPublic(WOLFTPM2_DEV* dev, word32 nvIndex,
 #endif
 
     if (nvPublic) {
-        XMEMCPY(nvPublic, &out.nvPublic.nvPublic, sizeof(*nvPublic));
+        nvPublic->attributes = out.nvPublic.nvPublic.attributes;
+        nvPublic->authPolicy.size = out.nvPublic.nvPublic.authPolicy.size;
+        if (nvPublic->authPolicy.size > 0) {
+            if (nvPublic->authPolicy.size > 
+                    (UINT16)sizeof(nvPublic->authPolicy.buffer)) {
+                nvPublic->authPolicy.size = 
+                    (UINT16)sizeof(nvPublic->authPolicy.buffer);
+            }
+            XMEMCPY(nvPublic->authPolicy.buffer,
+                    out.nvPublic.nvPublic.authPolicy.buffer,
+                    nvPublic->authPolicy.size);
+        }
+        nvPublic->dataSize = out.nvPublic.nvPublic.dataSize;
```

Comment:
Should this whole if be another helper function?

Reply:
I thought about it, but currently only used in one place. I will do anyways.